catroid/src/main/java/org/catrobat/catroid/ui/ProjectUtils.kt

```diff
@@ -0,0 +1,130 @@
+    val startListeningBrickExists = any { brick ->
+        brick is StartListeningBrick
+    }
+    val backgroundRequestOrWebRequestBrickExists = any { brick ->
+        brick is WebRequestBrick || brick is BackgroundRequestBrick || brick is LookRequestBrick
+    }
+    return startListeningBrickExists and backgroundRequestOrWebRequestBrickExists
+}
+
+/**
+ * extension boolean function for Project data type.
+ * check if the project contains black-listed bricks
+ * */
+private fun Project.shouldDisplayWarning(): Boolean {
+    val brickList = arrayListOf<Brick>()
+    sceneList.forEach { scene ->
+        brickList.run {
+            scene.backgroundSprite.allBricks.forEach {
+                add(it)
+                if (it is ForeverBrick) {
+                    addAll(it.nestedBricks)
+                    add(it.allParts.last())
+                }
+            }
+            scene.spriteList.forEach { sprite ->
+                sprite.allBricks.forEach {
+                    add(it)
+                    if (it is ForeverBrick) {
+                        addAll(it.nestedBricks)
+                        add(it.allParts.last())
+                    }
+                }
+            }
+        }
+    }
+    return brickList.containsBlackListedBricks()
+}
+
+/**
+ * [only once]!
+ * show warning dialog for downloaded projects which contains black-listed bricks
+ * */
+
+fun showWarningDialogIfNeeded(context: Context) {
+    // used for not showing the dialog again
+    val sharedPreferences = context.getSharedPreferences(
+        context.getString(R.string.preference_approved_list_file_key),
+        MODE_PRIVATE
+    )
+    val currentProject = ProjectManager.getInstance().currentProject
+    val projectUrl = currentProject.xmlHeader.remixParentsUrlString
+    // if project has an url => is a downloaded project
+    val isDownloadedProject = projectUrl.isNotBlank()
+    // since the projectUrl is kinda unique, ues it as key for the shared preference, if it's null
+    // that means the dialog hasn't been displayed yet
+    val showForFirstTime = sharedPreferences.getString(projectUrl, null).isNullOrBlank()
+
+    if (isDownloadedProject && currentProject.shouldDisplayWarning() && showForFirstTime) {
+        AlertDialog.Builder(context)
+            .setTitle(context.resources.getString(R.string.warning))
+            .setMessage(context.resources.getString(R.string.security_warning_dialog_msg))
+            .setCancelable(false)
+            .setPositiveButton(context.resources.getString(R.string.ok)) { dialog, _ ->
+                sharedPreferences
+                    .edit()
+                    .putString(projectUrl, projectUrl)
+                    .apply()
+                dialog.dismiss()
+            }
+            .show()
+    } else if (isDownloadedProject) {
+        // add it anyway to avoid showing this dialog for downloaded projects, but have
+        // black-listed bricks added by the user afterwards..
```

Comment:
This is not clear to me. Why would that be necessary? Isn't that executed only once anyway?

Reply:
this scenario will explain why is this needed:
1- download a project which doesn't have the combination of 2 suspicious bricks, 
2- modify the project by adding suspicious bricks
3- go to the Projects screen 
4- click on the downloaded modified project
5- the dialog is shown, which is not the case!